mergJSON/mergJSONLibrary.livecodescript

```diff
@@ -0,0 +1,94 @@
+﻿script "ws.goulding.script-library.mergjson"
+/**
+
+Name: mergJSON Support Library
+
+Summary:
+Support functions for the mergJSON external
+
+Requires:
+
+- mergJSON
+
+*/
+
+on revLoadLibrary
+   if the target is not me then
+      pass revLoadLibrary
+   end if
+   
+   insert the script of me into back
+   
+   if the environment contains "development" then
+      set the _ideoverride of me to true
+   end if
+end revLoadLibrary
+
+on revUnloadLibrary
+   if the target is not me then
+      pass revUnloadLibrary
+   end if
+   
+   remove the script of me from back
+end revUnloadLibrary
+
+/**
+
+Parse JSON to a LiveCode array
+
+Parameters:
+pJSON (string): A UTF8 encoded JSON string
+
+Returns: A LiveCode array
+
+Description:
+JSON arrays are translated to LiveCode numerically indexed arrays 
+with keys 1...N and JSON objects are translated to LiveCode array
+key -> value pairs.
+
+*/
+
+function JSONToArray pJSON
+   local tArray,tKeys
+   repeat for each line tKey in mergJSONDecode(pJSON,"tArray")
+      put JSONToArray(tArray[tKey]) into tArray[tKey]
+   end repeat
+   
+   if tArray is an array then
+      return tArray
+   else
+      return textDecode(tArray,"UTF8")
+   end if
+end JSONToArray
+
+/**
+
+Parse a LiveCode array to JSON
+
+Parameters:
+pArray (array): A LiveCode array
+pForceRootType (optional enum): Provides the ability to force an element
+to be an object even if it appears to be a numerically indexed array or
+a string even if it appears to be a number.
+
+Returns: A UTF8 encoded JSON string
+
+Description:
+By default LiveCode numerically indexed arrays with keys 1...N are
+translated to JSON arrays and LiveCode array key -> value pairs are
+translated to JSON objects.
+
+*/
+
+function ArrayToJSON pArray,pForceRootType,pPretty
+   repeat for each key tKey in pArray
+      if pArray[tKey] is an array then
+         put "}"&ArrayToJSON(pArray[tKey],pForceRootType) into pArray[tKey]
+      else if char 1 of pArray[tKey] = "}" then
+         put "}}" before pArray[tKey]
```

Comment:
This is something we had to do to work around the fact that the old externals API can't access multi-dimensional arrays. It's only values not keys. Values beginning with `}` are presumed to already be JSON. Values beginning with `}}` are forced to be strings. If you want to begin strings with those the main quirk is you will lose the `}`.